acceptance/lib/pxp-agent/test_helper.rb

```diff
@@ -394,3 +394,112 @@ def have_all_rpc_responses?(targets, responses)
   end
   return true
 end
+
+# A suite of functions for creating a manifest that sleeps for a long period
+# and interacting with Puppet runs of that manifest.
+def create_sleep_manifest(master, manifest, seconds_to_sleep)
+    create_remote_file(master, manifest, <<-MODULEPP)
+node default {
+  case $::osfamily {
+    'windows': { exec { 'sleep': command => 'C:/Windows/system32/ping.exe 127.0.0.1 -n #{seconds_to_sleep}', returns => 1 } }
+    'Darwin':  { exec { 'sleep': command => '/bin/sleep #{seconds_to_sleep} || /usr/bin/true', } }
+    default:   { exec { 'sleep': command => '/bin/sleep #{seconds_to_sleep} || /bin/true', } }
+  }
+}
+MODULEPP
+    on(master, "chmod 644 #{manifest}")
+end
+
+def wait_for_sleep_process(target)
+  begin
+    ps_cmd = target['platform'] =~ /win/ ? 'ps -efW' : 'ps -ef'
+    sleep_process = target['platform'] =~ /win/ ? 'PING' : '/bin/sleep'
+    retry_on(target, "#{ps_cmd} | grep '#{sleep_process}' | grep -v 'grep'", {:max_retries => 15, :retry_interval => 2})
+  rescue
+    fail("After triggering a puppet run on #{target} the expected sleep process did not appear in process list")
+  end
+end
+
+def stop_sleep_process(targets)
+  targets = [targets].flatten
+
+  targets.each do |target|
+    case target['platform']
+    when /osx/
+      command = "ps -e -o pid,comm | grep sleep | sed 's/^[^0-9]*//g' | cut -d\\  -f1"
+    when /win/
+      command = "ps -efW | grep PING | sed 's/^[^0-9]*[0-9]*[^0-9]*//g' | cut -d ' ' -f1"
+    else
+      command = "ps -ef | grep 'bin/sleep ' | grep -v 'grep' | grep -v 'true' | sed 's/^[^0-9]*//g' | cut -d\\  -f1"
```

Comment:
I'm not really sure, I carried this over from what `restart_pxp_agent_during_non_blocking_run` was doing before.